exoctk/atmospheric_retrievals/aws_tools.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python
+
+"""This module contains various functions for interacting with AWS for
+``exoctk`` atmospheric retrievals.
+
+Authors
+-------
+
+    - Matthew Bourque
+
+Use
+---
+
+    This script is inteneded to be imported and used by other modules,
+    for example:
+
+        from aws_tools import get_config
+        get_config()
+
+Dependencies
+------------
+
+    Dependent libraries include:
+
+    - boto3
+    - paramiko
+    - scp
+
+    Users must also have a ``aws_config.json`` file present within the
+    ``atmospheric_retrievals`` subdirectory.  This file must be of
+    a valid JSON format and contain two key/value pairs,
+    ``ec2_id`` and ``ssh_file``, e.g.:
+
+    {
+    "ec2_id" : "lt-021de8b904bc2b728",
+    "ssh_file" : "~/.ssh/my_ssh_key.pem"
+    }
+
+    where the ``ec2_id`` contains the ID for an EC2 launch template
+    or an existing EC2 instance, and ``ssh_file`` points to the SSH
+    public key used for logging into an AWS account.
+"""
+
+import json
+import logging
+import os
+import time
+
+import boto3
+import paramiko
+from scp import SCPClient
+
+
+def build_environment(instance, key, client):
+    """Builds an ``exoctk`` environment on the given AWS EC2 instance
+
+    Parameters
+    ----------
+    instance : obj
+        A ``boto3`` AWS EC2 instance object.
+    key : obj
+        A ``paramiko.rsakey.RSAKey`` object.
+    client : obj
+        A ``paramiko.client.SSHClient`` object.
+    """
+
+    logging.info('Building ExoCTK environment')
+
+    # Connect to the EC2 instance and run commands
+    connected = False
+    iterations = 0
+    while not connected:
+        if iterations == 12:
+            break
```

Comment:
Do you want to log a warning here if the break has to occur because it took too long to connect? (I think you do this below in the transfer function and that seems useful to a user to me.)

Reply:
Good idea!